programs/stake/src/stake_state.rs

```diff
@@ -6409,6 +6455,236 @@ mod tests {
         assert_eq!(new_stake.delegation.stake, delegation * 2);
     }
 
+    /// Ensure that `initialize()` respects the MINIMUM_STAKE_DELEGATION requirements
+    /// - Assert 1: accounts with a balance equal-to the minimum initialize OK
+    /// - Assert 2: accounts with a balance less-than the minimum do not initialize
+    #[test]
+    fn test_initialize_minimum_stake_delegation() {
+        for (stake_delegation, expected_result) in [
+            (MINIMUM_STAKE_DELEGATION, Ok(())),
+            (
+                MINIMUM_STAKE_DELEGATION - 1,
+                Err(InstructionError::InsufficientFunds),
+            ),
+        ] {
+            let rent = Rent::default();
+            let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
+            let stake_pubkey = Pubkey::new_unique();
+            let stake_account = AccountSharedData::new_ref(
+                stake_delegation + rent_exempt_reserve,
+                std::mem::size_of::<StakeState>(),
+                &id(),
+            );
+            let stake_keyed_account = KeyedAccount::new(&stake_pubkey, false, &stake_account);
+
+            assert_eq!(
+                expected_result,
+                stake_keyed_account.initialize(
+                    &Authorized::auto(&stake_pubkey),
+                    &Lockup::default(),
+                    &rent
+                ),
+            );
+        }
+    }
+
+    /// Ensure that `delegate()` respects the MINIMUM_STAKE_DELEGATION requirements
+    /// - Assert 1: delegating an amount equal-to the minimum delegates OK
+    /// - Assert 2: delegating an amount less-than the minimum do not delegate
+    /// Also test both asserts above over both StakeState::{Initialized and Stake}, since the logic
+    /// is slightly different for the variants.
+    #[test]
+    fn test_delegate_minimum_stake_delegation() {
+        for (stake_delegation, expected_result) in [
+            (MINIMUM_STAKE_DELEGATION, Ok(())),
+            (
+                MINIMUM_STAKE_DELEGATION - 1,
+                Err(InstructionError::StakeDelegationTooSmall),
+            ),
+        ] {
+            let rent = Rent::default();
+            let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
+            let stake_pubkey = Pubkey::new_unique();
+            let signers = HashSet::from([stake_pubkey]);
+            let meta = Meta {
+                rent_exempt_reserve,
+                ..Meta::auto(&stake_pubkey)
+            };
+
+            for stake_state in &[
+                StakeState::Initialized(meta),
+                StakeState::Stake(meta, Stake::default()),
```

Comment:
Yeah... I wasn't sure on this either, since in theory it shouldn't be possible. I'll go ahead and remove it, since `delegate()` already has code to prevent this.

Reply:
Done!